### PR TITLE
DataCubeViewer tweaks, camera class error fix, makefile improvements

### PR DESCRIPTION
**Pull request by christhechris**
- Tweaks to visualiser
- add label options to DataCubeViewer, roll back some changes
- add data folders to gitignore file
- remove try-except around imports, allows error to be raised instead of turning into warning. Previous behaviour resulted in errors when missing import is used later in class, leading to unlcear error messages.
- add conda stuff to Makefile, move some requirements to dev requirement in settings.ini, bump version to 0.2.2

### CODE REVIEW

File: settings.ini [hunk 1]
```diff
@@ -8,14 +8,17 @@ author = Yiwei Mao, et al.
...
 audience = Developers
 language = English
 custom_sidebar = True
 license = apache2
 status = 4
-requirements = numpy scipy matplotlib nbdev fastcore fastprogress h5py xarray pillow pandas tqdm bokeh holoviews panel param Py6S pyserial netcdf4 streamz pyviz_comms hvplot requests<=2.24.0 datashader psutil
+requirements = numpy scipy matplotlib fastcore fastprogress h5py xarray pillow pandas tqdm bokeh holoviews panel param Py6S pyserial netcdf4 streamz pyviz_comms hvplot requests<=2.24.0 datashader psutil
+pip_requirements = opencv-python
+conda_requirements = 
+dev_requirements = nbdev conda-build twine
```
Comment (YiweiMao) [inline settings.ini:21]: I think we should remove opencv-python from pip_requirements. It was really only for testing with webcams - not used in `SimulatedCamera`. This format separating the pip and conda makes sense though and I like it.

Comment (christhechris) [reply settings.ini:21]: done.

**Comment by christhechris** [comment]
@YiweiMao did you want to put fix for index.html in this PR?  I could not see the cause on my quick look.

**Comment by YiweiMao** [comment]
> @YiweiMao did you want to put fix for index.html in this PR? I could not see the cause on my quick look.
Yes, I only got to this just now. I'll put my fixes in after understanding what changed.

**Comment by christhechris** [comment]
> @YiweiMao did you want to put fix for index.html in this PR? I could not see the cause on my quick look.
> Yes, I only got to this just now. I'll put my fixes in after understanding what changed.

OK, it is related to how the openhsi_cam.png is included. nbdev seems to not like relative path I think?

**Comment by YiweiMao** [comment]
Okay, docs homepage should be fixed now. And I updated the DataCubeViewer to handle the new box size edge cases. Caps the indexing to the image size in case it goes over (or < 0). 
```python
x_slice = slice( np.max((x-self._bhalf[1],0)), np.min((x-self._bhalf[1]+self.box_sz[1],self.dc.dc.data.shape[1])) )
y_slice = slice( np.max((y-self._bhalf[0],0)), np.min((y-self._bhalf[0]+self.box_sz[0],self.dc.dc.data.shape[0])) )
c = np.mean(self.dc.dc.data[y_slice,x_slice,:],axis=(0,1))
```

**Pull request merged**